flink-formats/flink-orc/src/main/java/org/apache/flink/orc/OrcFileFormatFactory.java

```diff
@@ -167,5 +190,187 @@ public ChangelogMode getChangelogMode() {
+
+        private ColumnStats convertToColumnStats(
+                long totalRowCount, LogicalType logicalType, ColumnStatistics columnStatistics) {
+            ColumnStats.Builder builder =
+                    new ColumnStats.Builder().setNdv(null).setAvgLen(null).setMaxLen(null);
+            if (!columnStatistics.hasNull()) {
+                builder.setNullCount(0L);
+            } else {
+                builder.setNullCount(totalRowCount - columnStatistics.getNumberOfValues());
+            }
+
+            switch (logicalType.getTypeRoot()) {
+                case BOOLEAN:
+                    break;
+                case TINYINT:
+                case SMALLINT:
+                case INTEGER:
+                case BIGINT:
+                    if (columnStatistics instanceof IntegerColumnStatistics) {
+                        builder.setMax(((IntegerColumnStatistics) columnStatistics).getMaximum())
+                                .setMin(((IntegerColumnStatistics) columnStatistics).getMinimum());
+                        break;
+                    } else {
+                        return null;
+                    }
+                case FLOAT:
+                case DOUBLE:
+                    if (columnStatistics instanceof DoubleColumnStatistics) {
+                        builder.setMax(((DoubleColumnStatistics) columnStatistics).getMaximum())
+                                .setMin(((DoubleColumnStatistics) columnStatistics).getMinimum());
+                        break;
+                    } else {
+                        return null;
+                    }
+                case CHAR:
+                case VARCHAR:
+                    if (columnStatistics instanceof StringColumnStatistics) {
+                        builder.setMax(((StringColumnStatistics) columnStatistics).getMaximum())
+                                .setMin(((StringColumnStatistics) columnStatistics).getMinimum());
+                        break;
+                    } else {
+                        return null;
+                    }
+                case BINARY:
+                case VARBINARY:
+                    if (columnStatistics instanceof StringColumnStatistics) {
+                        break;
+                    } else {
+                        return null;
+                    }
+                case DATE:
+                    if (columnStatistics instanceof DateColumnStatistics) {
+                        Date maximum =
+                                (Date) ((DateColumnStatistics) columnStatistics).getMaximum();
+                        Date minimum =
+                                (Date) ((DateColumnStatistics) columnStatistics).getMinimum();
+                        builder.setMax(maximum).setMin(minimum);
+                        break;
+                    } else {
+                        return null;
+                    }
+                case TIMESTAMP_WITHOUT_TIME_ZONE:
+                case TIME_WITHOUT_TIME_ZONE:
```

Comment:
`TIME_WITHOUT_TIME_ZONE` should be considered alone

Reply:
> `TIME_WITHOUT_TIME_ZONE` should be considered alone

Ok. Btw, orc don't support TIMESTAMP_WITHOUT_LOCAL_TIME_ZONE.